### ISSUE TITLE
Can't Remove Layer "treemacs" and Its Packages

### ISSUE THREAD

**Issue by Johnson9009** [syl20bnr/spacemacs#13595]
I want to remove the layers and packages that I haven't used, because my work environment have a terrible network, so I want simply my spacemacs packages.

I don't need and use file tree like "treemacs", "neotree", so I want to remove them from my spacemacs dependent, I haven't added them in  `dotspacemacs-configuration-layers`, but I found the "treemacs" is fixed enabled inside "+distributions/spacemacs".

https://github.com/syl20bnr/spacemacs/blob/develop/layers/%2Bdistributions/spacemacs/layers.el#L19

Even through adding "treemacs" into "dotspacemacs-excluded-packages", the packages used by layer "treemacs" still will be downloaded.

Why does `helm` and `treemacs` are forced added inside `+distributions/spacemacs`, for the users who don't use `helm`(they maybe just use ivy) and any of file tree, it is unreasonable.

And how can I remove "treemacs" layer?
Thanks.

**Comment by smile13241324** [comment]
This is misunderstanding @Johnson9009. Basically spacemacs requires a search solution as well as a tree solution. However for both you can choose between nice and shiny or simple and fast.

For searching you can choose between helm and ivy, for trees between treemacs and neotree. If no decision has been made, aka non of these layers are declared, spacemacs is pulling what it prefers which is treemacs and helm at the moment.

If you do not want any search solution or file tree at all you can still switch distributions from spacemacs to spacemacs-base where you have full control.

**Comment by Johnson9009** [comment]
@smile13241324 Thanks for your detailed explanation, I use Spacemacs almost everyday, its excellent features help me a lot.

Long time ago I have found the "treemacs" feature of Spacemacs, at the first glance I think it is a very useful layer and I think I will need to use it. But I haven't use it until now, the "counsel-find-file" is good enough so I can't remeber to use "treemacs", so now I start to think whether I really need the "filetree" like "neotree", "treemacs".

I agree Spacemacs need a search solution, if without helm or ivy I think I can't use emacs efficiently, but whether Spacemacs must need a tree solution especially for terminal users? The "counsel-find-file" and "ivy-switch-buffer" already cover lots of functions of "filetree".

So I think the "treemacs" shouldn't be forced added inside distribution of "spacemacs", is there any reason why Spacemacs must need a tree solution?

Thanks a lot.

**Comment by Alexander-Miller** [comment]
> So I think the "treemacs" shouldn't be forced added inside distribution of "spacemacs", is there any reason why Spacemacs must need a tree solution?

Probably because every other editor (other than vim) has a file tree too, so it's pretty much a standard component.

Switching distributions is not necessary either, you can use `dotspacemacs-excluded-packages`. 
However treemacs is also used as a render backend for various tree views in lsp-ui, so removing it completely might give you trouble later.

> However for both you can choose between nice and shiny or simple and fast.

Gonna have to object here: treemacs is an order of magnitude faster than neotree (though of course you'll only notice when you have a couple hundred or thousand files to display).

**Comment by smile13241324** [comment]
@Alexander-Miller you have got me there :), treemacs is not only nicer but also faster. My point was rather that some solutions come with more functionality and others with less.

@Johnson9009 I don't know why technically a tree solution is required, I would still assume that most users will expect a tree solution being installed already. I have asked the other maintainers maybe they can tell you more.

Nevertheless I don't think that there will be a change, as I assume that 90% of all users just expect a tree solution, especially when LSP is used, which is almost everywhere these days, treemacs is used internally.

**Comment by Johnson9009** [comment]
@Alexander-Miller @smile13241324 Thanks for your nice reply, I have got your point, yes from the point view of modern editor the file tree solution is an essential feature. 

I think the reason why I want to remove the "treemacs" is I only have used little part features of Spacemacs, e.g., I haven't used lsp and even projectile yet, I also think lsp is the essential feature to use, when I have time I will try to use lsp replace my anaconda-mode for Python completion, maybe at that time I will need the tree solution.

@Alexander-Miller Thanks for the suggestion of `dotspacemacs-excluded-packages`, I have tried it before, but I found all of the packages must be added in `dotspacemacs-excluded-packages`, if only the `treemacs` is added, then any package of layer `treemacs` won't be uninstalled.
At this time, I found a better way to uninstall layer `treemacs` in `~/.spacemacs.d/init.el`.
```elisp
(defun dotspacemacs/layers ()
  (setq-default 
   dotspacemacs-configuration-layers
   '(
     (treemacs :packages nil)
    )))
```
